platform/android/plugin/godot_plugin_jni.cpp

```diff
@@ -114,16 +114,19 @@ JNIEXPORT void JNICALL Java_org_godotengine_godot_plugin_GodotPlugin_nativeEmitS
 	String signal_name = jstring_to_string(j_signal_name, env);
 
 	int count = env->GetArrayLength(j_signal_params);
-	const Variant *args[count];
+	Variant args[VARIANT_ARG_MAX];
 
-	for (int i = 0; i < count; i++) {
+	ERR_FAIL_COND_MSG(count > VARIANT_ARG_MAX, "Maximum argument count exceeded!")
+
+	for (int i = 0; i < MIN(count, VARIANT_ARG_MAX); i++) {
 		jobject j_param = env->GetObjectArrayElement(j_signal_params, i);
-		Variant variant = _jobject_to_variant(env, j_param);
-		args[i] = &variant;
+		if (j_param) {
+			args[i] = _jobject_to_variant(env, j_param);
+		}
 		env->DeleteLocalRef(j_param);
 	};
 
-	singleton->emit_signal(signal_name, args, count);
+	singleton->emit_signal(signal_name, args[0], args[1], args[2], args[3], args[4]);
```

Comment:
I'm thinking that a static assert here would make it safer since the number of arguments is hardcoded:
`static_assert(VARIANT_ARG_MAX == 5);`

Reply:
I'll make a separate PR for this to add this to `call_deferred`, too.